grpc/src/main/java/io/stargate/grpc/service/EnrichedQueryHandler.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.grpc.service;
+
+import com.datastax.oss.driver.api.core.ProtocolVersion;
+import io.grpc.Status;
+import io.stargate.db.ClientInfo;
+import io.stargate.db.ImmutableParameters;
+import io.stargate.db.PagingPosition;
+import io.stargate.db.Parameters;
+import io.stargate.db.Persistence;
+import io.stargate.db.Persistence.Connection;
+import io.stargate.db.Result;
+import io.stargate.db.Result.Prepared;
+import io.stargate.db.RowDecorator;
+import io.stargate.db.datastore.ArrayListBackedRow;
+import io.stargate.db.datastore.Row;
+import io.stargate.db.schema.Column;
+import io.stargate.db.schema.TableName;
+import io.stargate.grpc.service.GrpcService.EnrichedResponseAndTraceId;
+import io.stargate.proto.BridgeQuery.EnrichedResponse;
+import io.stargate.proto.QueryOuterClass.Query;
+import io.stargate.proto.QueryOuterClass.QueryParameters;
+import io.stargate.proto.QueryOuterClass.SchemaChange;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ScheduledExecutorService;
+import org.apache.cassandra.stargate.db.ConsistencyLevel;
+
+public abstract class EnrichedQueryHandler extends EnrichedMessageHandler<Query, Prepared> {
```

Comment:
In the chain above, `ResponseAndTraceId` is not an instance of `GeneratedMessageV3` so it can't be the `ReturnT` of BaseQueryHandler...but I think even without that there are other methods in MessageHandler that return a Response.Builder type that can't be made generic.

Really I need `ResultT` to extend "something that's guaranteed to have a Builder static class on it", if that makes sense - and I don't think there is such a type in the autogenerated code, which is what makes this so difficult to make generic. There are a couple of places outside the diff, such as
`protected Response.Builder makeResponseBuilder(Result result)` and the definition of `ResponseAndTraceId` that would require either a generic as described above, or a generic on the Builder type, or maybe both with some sort of unsafe casting solution.